lib/secretsscanner/scan/scan_test.go

```diff
@@ -0,0 +1,192 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package scan
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"sort"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/crypto/ssh"
+	cryptosshtestdata "golang.org/x/crypto/ssh/testdata"
+	"google.golang.org/protobuf/testing/protocmp"
+
+	accessgraphsecretsv1pb "github.com/gravitational/teleport/api/gen/proto/go/teleport/accessgraph/v1"
+	"github.com/gravitational/teleport/api/types/accessgraph"
+)
+
+var (
+	deviceID = uuid.NewString()
+)
+
+func TestNewScanner(t *testing.T) {
+	tests := []struct {
+		name    string
+		keysGen func(t *testing.T, path string) []*accessgraphsecretsv1pb.PrivateKey
+	}{
+		{
+			name:    "encrypted keys",
+			keysGen: writeEncryptedKeys,
+		},
+		{
+			name:    "unencrypted keys",
+			keysGen: writeUnEncryptedKeys,
+		},
+		{
+			name:    "encryptedKey without public key file",
+			keysGen: writeEncryptedKeyWithoutPubFile,
+		},
```

Comment:
Let's also test:

- Files with headers that don't match "supportedPrivateKeyHeaders"
- Files with garbage before the headers
- Files with less than 150 bytes in length
- Files with proper headers but invalid keys inside